MCU_Aurix/bootloader/src/uds.c

```diff
@@ -25,55 +25,19 @@
 
 static isoTP* iso;
 
-// TODO should I add these UDS_Msg functions to the header?
-struct UDS_Msg {
-    uint32_t len;
-    uint8_t data[]; // flexible array member
-};
-typedef struct UDS_Msg UDS_Msg;
-
-uint8_t getSID(UDS_Msg *msg){
-    if (msg->len == 0){
-        return 0;
-    }
-    uint8_t first_byte = msg->data[0];
-    return (first_byte & 0x3f); // TODO 6 Bit of first byte are SID, right?
-}
-
-uint16_t getDID(UDS_Msg *msg){
-    if (msg->len == 0){
-        return 0;
-    }
-    uint8_t high = msg->data[1];                              // DID high Byte
-    uint8_t low = msg->data[2];                               // DID low Byte
-    uint16_t did = ((uint16_t) high) << 8;
-    did |= low;
-    return did;
-}
-
-uint32_t getMemoryAddress(UDS_Msg *msg){
-    if (msg->len == 0){
-        return 0;
-    }
-    uint32_t addr = 0;
-    addr |= (msg->data[1] << 24);
-    addr |= (msg->data[2] << 16);
-    addr |= (msg->data[3] << 8);
-    addr |= msg->data[4];
+//============================================================================
+// Private Helper Function declarations
+//============================================================================
 
-    return addr;
-}
+uint8_t getSID(UDS_Msg *msg);
+uint16_t getDID(UDS_Msg *msg);
+uint32_t getMemoryAddress(UDS_Msg *msg);
+uint16_t getNoBytes(UDS_Msg *msg);
```

Comment:
Haetten wir eig direkt static machen sollen.
```suggestion
static uint8_t getSID(UDS_Msg *msg);
static uint16_t getDID(UDS_Msg *msg);
static uint32_t getMemoryAddress(UDS_Msg *msg);
static uint16_t getNoBytes(UDS_Msg *msg);
```